### PR TITLE
(jkube-kit/enricher/generic/) : Migrate tests from JMockit to Mockito

### PR DESCRIPTION
**Pull request by baruKreddy**
Fixes https://github.com/eclipse/jkube/issues/1591

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenIssueManagementEnricherTest.java [hunk 1]
```diff
@@ -19,25 +19,28 @@
...
-import mockit.Mocked;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.util.Map;
 
 import static junit.framework.TestCase.assertEquals;
 import static junit.framework.TestCase.assertNotNull;
 import static junit.framework.TestCase.assertTrue;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenIssueManagementEnricherTest.java:30]:
```suggestion

```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenProjectEnricherTest.java [hunk 2]
```diff
@@ -34,23 +31,22 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenProjectEnricherTest.java:33]:
```suggestion
```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java [hunk 3]
```diff
@@ -19,24 +19,28 @@
...
-import mockit.Expectations;
-import mockit.Mocked;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.util.Map;
 
 import static junit.framework.TestCase.assertEquals;
 import static junit.framework.TestCase.assertNotNull;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java:29]:
```suggestion

```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java [hunk 4]
```diff
@@ -33,13 +30,12 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java:32]:
```suggestion

```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java [hunk 5]
```diff
@@ -78,27 +68,18 @@ public void testMavenScmOnlyConnection() {
...
 
         Map<String, String> scmAnnotations = builder.buildFirstItem().getMetadata().getAnnotations();
-        assertNotNull(scmAnnotations);
 
-        Assert.assertEquals(1, scmAnnotations.size());
-        Assert.assertFalse(scmAnnotations.containsKey(JKubeAnnotations.SCM_TAG.value()));
-        Assert.assertEquals("scm:git:git://github.com/jkubeio/kubernetes-maven-plugin.git", scmAnnotations.get(JKubeAnnotations.SCM_URL.value()));
+        assertThat(scmAnnotations).isNotNull().hasSize(1)
+                .containsEntry(JKubeAnnotations.SCM_URL.value(),"scm:git:git://github.com/jkubeio/kubernetes-maven-plugin.git");
+        assertThat(scmAnnotations.containsKey(JKubeAnnotations.SCM_TAG.value())).isFalse();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java:82]: These two assertions can be chained together:
```suggestion
        assertThat(scmAnnotations).isNotNull().hasSize(1)
            .containsEntry(JKubeAnnotations.SCM_URL.value(),"scm:git:git://github.com/jkubeio/kubernetes-maven-plugin.git")
            .doesNotContainKey(JKubeAnnotations.SCM_TAG.value());
```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java [hunk 6]
```diff
@@ -109,28 +90,17 @@ public void testMavenScmOnlyDevConnection() {
...
         Map<String, String> scmAnnotations = builder.buildFirstItem().getMetadata().getAnnotations();
-        assertNotNull(scmAnnotations);
-
-        Assert.assertEquals(1, scmAnnotations.size());
-        assertEquals("git://github.com/jkubeio/kubernetes-maven-plugin.git",
-                scmAnnotations.get(JKubeAnnotations.SCM_URL.value()));
-        Assert.assertFalse(scmAnnotations.containsKey(JKubeAnnotations.SCM_TAG.value()));
+        assertThat(scmAnnotations).isNotNull().hasSize(1)
+                .containsEntry(JKubeAnnotations.SCM_URL.value(),"git://github.com/jkubeio/kubernetes-maven-plugin.git");
+        assertThat(scmAnnotations.containsKey(JKubeAnnotations.SCM_TAG.value())).isFalse();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java:103]: You can chain these two assertions together

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java [hunk 7]
```diff
@@ -140,51 +110,30 @@ public void testMavenScmOnlyUrl() {
...
 
         Map<String, String> scmAnnotations = builder.buildFirstItem().getMetadata().getAnnotations();
-        assertNotNull(scmAnnotations);
 
-        Assert.assertEquals(1, scmAnnotations.size());
-        Assert.assertFalse(scmAnnotations.containsKey(JKubeAnnotations.SCM_TAG.value()));
-        Assert.assertEquals("scm:git:git://github.com/jkubeio/kubernetes-maven-plugin.git", scmAnnotations.get(JKubeAnnotations.SCM_URL.value()));
+        assertThat(scmAnnotations).isNotNull().hasSize(1)
+                .containsEntry(JKubeAnnotations.SCM_URL.value(),"scm:git:git://github.com/jkubeio/kubernetes-maven-plugin.git");
+        assertThat(scmAnnotations.containsKey(JKubeAnnotations.SCM_TAG.value())).isFalse();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenScmEnricherTest.java:124]: You can chain these assertions together

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java [hunk 8]
```diff
@@ -77,22 +69,24 @@ public void testCustomAppName() {
     KubernetesList list = builder.build();
 
     Map<String, String> labels = list.getItems().get(0).getMetadata().getLabels();
+    assertThat(labels.get("project")).isNull();
+
+    assertThat(labels).isNotNull()
+            .containsEntry("group","groupId")
+            .containsEntry("app","my-custom-app-name")
+            .containsEntry("version","version");
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java:77]:
```suggestion
    assertThat(labels).isNotNull()
            .doesNotContainKey("project")
            .containsEntry("group","groupId")
            .containsEntry("app","my-custom-app-name")
            .containsEntry("version","version");
```

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java [hunk 9]
```diff
@@ -77,22 +69,24 @@ public void testCustomAppName() {
...
     Map<String, String> selectors = deployment.getSpec().getSelector().getMatchLabels();
-    assertEquals("groupId", selectors.get("group"));
-    assertEquals("my-custom-app-name", selectors.get("app"));
-    assertNull(selectors.get("version"));
-    assertNull(selectors.get("project"));
+    assertThat(selectors)
+            .containsEntry("group","groupId")
+            .containsEntry("app","my-custom-app-name");
+    assertThat(selectors.get("version")).isNull();
+    assertThat(selectors.get("project")).isNull();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java:89]: Chain these assertions together

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java [hunk 10]
```diff
@@ -106,21 +100,23 @@ public void testEmptyCustomAppName() {
...
-    assertEquals("", labels.get("app"));
-    assertEquals("version", labels.get("version"));
-    assertNull(labels.get("project"));
 
+    assertThat(labels.get("project")).isNull();
+    assertThat(labels)
+            .isNotNull()
+            .containsEntry("group","groupId")
+            .containsEntry("app","")
+            .containsEntry("version","version");
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java:109]: please chain these assertions together

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java [hunk 11]
```diff
@@ -106,21 +100,23 @@ public void testEmptyCustomAppName() {
...
     Map<String, String> selectors = deployment.getSpec().getSelector().getMatchLabels();
-    assertEquals("groupId", selectors.get("group"));
-    assertEquals("", selectors.get("app"));
-    assertNull(selectors.get("version"));
-    assertNull(selectors.get("project"));
+    assertThat(selectors)
+            .containsEntry("group","groupId")
+            .containsEntry("app","");
+    assertThat(selectors.get("version")).isNull();
+    assertThat(selectors.get("project")).isNull();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java:119]: these assertions can be chained too

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java [hunk 12]
```diff
@@ -131,21 +127,22 @@ public void testDefaultAppName() {
...
 
-    assertNotNull(labels);
-    assertEquals("groupId", labels.get("group"));
-    assertEquals("artifactId", labels.get("app"));
-    assertEquals("version", labels.get("version"));
-    assertNull(labels.get("project"));
+    assertThat(labels).isNotNull()
+            .containsEntry("group","groupId")
+            .containsEntry("app","artifactId").containsEntry("version","version");
+    assertThat(labels.get("project")).isNull();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java:133]: chain these assertions together

File: jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java [hunk 13]
```diff
@@ -131,21 +127,22 @@ public void testDefaultAppName() {
...
-    assertEquals("groupId", selectors.get("group"));
-    assertEquals("artifactId", selectors.get("app"));
-    assertNull(selectors.get("version"));
-    assertNull(selectors.get("project"));
+    assertThat(selectors)
+            .containsEntry("group","groupId")
+            .containsEntry("app","artifactId");
+
+    assertThat(selectors.get("version")).isNull();
+    assertThat(selectors.get("project")).isNull();
```
Comment (rohanKanojia) [inline jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/ProjectLabelEnricherTest.java:145]: chain these assertions together

**Review by rohanKanojia — APPROVED** [review]
Minor comment regarding chaining assertions otherwise Looks good

**Comment by rohanKanojia** [comment]
@baruKreddy : Could you please resolve conflicts here?

**Review by manusa — APPROVED** [review]

**Comment by anurag-rajawat** [comment]
@baruKreddy please recheck AC before making your PR ready for review, especially regarding JUnit4 assertions.

**Comment by baruKreddy** [comment]
@anurag-rajawat Here do you think I made any mistake on assertions for Junit4 changes? Please point me out ,Just asking to know for not repeating it again!

**Comment by anurag-rajawat** [comment]
> @anurag-rajawat Here do you think I made any mistake on assertions for Junit4 changes?

No, but you forgot to replace JUnit's assertions with AssertJ

**Comment by anurag-rajawat** [comment]
> Please point me out

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java#L80-L88

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java#L115-L123

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java#L153-L163

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java#L194-L207

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/TriggersAnnotationEnricherTest.java#L236

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/RevisionHistoryEnricherTest.java#L85

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/MavenIssueManagementEnricherTest.java#L58

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java#L292

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java#L221

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java#L200

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java#L187

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/RouteEnricherTest.java#L172

https://github.com/eclipse/jkube/blob/8271c241d073fb95c5557e603dd83ce0d984024e/jkube-kit/enricher/generic/src/test/java/org/eclipse/jkube/enricher/generic/openshift/AutoTLSEnricherTest.java#L120-L149

**Comment by baruKreddy** [comment]
@anurag-rajawat Sorry I see , I missed couple of classes on assertions, Let me be more careful from next time, Thank you 👍

**Pull request merged**